codex/contracts/market.nim

```diff
@@ -91,7 +107,7 @@ method subscribeSlotFilled*(market: OnChainMarket,
                            Future[MarketSubscription] {.async.} =
   proc onEvent(event: SlotFilled) {.upraises:[].} =
     if event.requestId == requestId and event.slotIndex == slotIndex:
-      callback(event.requestId, event.slotIndex)
+      asyncSpawn callback(event.requestId, event.slotIndex)
```

Comment:
What happens if the `callback` fails? `asyncSpawn` is usually the wrong primitive to use in this construct, maybe `await callback`?

Reply:
Should be resolved with the new async state machine design.